Python 3.13 removed the pipes module, so switch to shlex for the same functionality.

Author: Jan Andre Reuter (JSC)

diff --color -Naur llvm-project-15.0.7.src.orig/llvm/test/Bindings/Go/lit.local.cfg llvm-project-15.0.7.src/llvm/test/Bindings/Go/lit.local.cfg
--- llvm-project-15.0.7.src.orig/llvm/test/Bindings/Go/lit.local.cfg	2023-01-12 08:12:30.000000000 +0100
+++ llvm-project-15.0.7.src/llvm/test/Bindings/Go/lit.local.cfg	2025-08-07 22:47:15.384884023 +0200
@@ -1,5 +1,4 @@
 import os
-import pipes
 import shlex
 import sys
 
@@ -56,7 +55,7 @@
     except (AttributeError, OSError):
         pass
 
-    return ' '.join([pipes.quote(arg) for arg in args])
+    return ' '.join([shlex.quote(arg) for arg in args])
 
 config.environment['CC'] = fixup_compiler_path(config.host_cc)
 config.environment['CXX'] = fixup_compiler_path(config.host_cxx)
